### PR TITLE
fix(findPotionResource): fix resource resolution

### PR DESCRIPTION
**Pull request by JohnSebastianHussey**

### CODE REVIEW

File: src/core/utils.spec.ts [hunk 1]
```diff
@@ -603,6 +604,21 @@ describe('potion/core', () => {
...
+                const resources = {
+                    '/foo': Foo,
+                    '/foo_bar': FooBar
+                };
+
+                expect(findPotionResource('/foo_bar/1', resources)).toEqual({
+                    resourceURI: '/foo_bar',
+                    resource: FooBar
+                });
+                expect(findPotionResource('/foo/1', resources)).toEqual({
```
Comment (rolandjitsu) [inline src/core/utils.spec.ts:617]: Could you add `findPotionResource('/foo', resources) -> Foo` and `findPotionResource('/foo_bar', resources) -> FooBar` as well?

File: src/core/utils.ts [hunk 2]
```diff
@@ -277,6 +277,9 @@ export function getPotionID(uri: string, resourceURI: string): PotionID {
 export function findPotionResource(uri: string, resources: PotionResources): {resourceURI: string, resource: typeof Item} | undefined {
     if (isString(uri)) {
         const entry = Object.entries(resources)
+            .sort((a, b) => {
```
Comment (rolandjitsu) [inline src/core/utils.ts]: You don't need the curly braces, you can `([a], [b]) => a.length - b.length`.

**Review by rolandjitsu — CHANGES_REQUESTED** [review]

File: src/core/utils.spec.ts (hunk 1 above)
Comment (JohnSebastianHussey) [reply src/core/utils.spec.ts:617]: Yes - will add these now.

File: src/core/utils.ts (hunk 2 above)
Comment (JohnSebastianHussey) [reply src/core/utils.ts]: Ok - I will change it.

**Review by rolandjitsu — APPROVED** [review]

**Pull request merged**